### ISSUE TITLE
A11y_AzureToolsStorageExplorer_BlobContainerTable_Table_ScreenReader: Voiceover does not read the table correctly as the row and column reference of the cell value.

### ISSUE THREAD

**Issue by nirmalsahoo** [microsoft/AzureStorageExplorer#1959]
[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”
GitHubTags: #A11y_AzureToolsStorageExplorer; #A11yMAS; #A11YTCS; #DesktopApp; #MAS1.3.1;Voiceover
**Environment Details:**
Desktop App: MAC OS Mojave
Version:10.14.6(18G29g)

**Steps to Reproduce:**
1. Launch Storage Explorer
2. Navigate to "Open Connect dialog" pane and Sign in to Azure.
3. Go to "Manage Accounts" pane and Select the subscriptions you will use. Once selected, click on "Apply" button.
4. Navigate to "Blob Containers" of the selected subscription, right click and hit enter on "Create Blob container" to Create a new Blob container.
5. Once completed, Navigate to "Upload" button and Upload a blob/file by using  "Upload Files" control.
6. Navigate to Table present in right side of the storage explorer.
7. Open Voiceover and Check whether the Voiceover reads the table cell value correctly.

**Actual:**
While navigating through keyboard, Voiceover does not read the table correctly as the row and column reference of the cell value.

**Same issue is observed throughout the application where the table is present.**

**Expected:**
Voiceover should read the table correctly that the row and column reference of the table cell value. 

**Recommendations:**
Refer below link which is repository of bug fixes code snippets:
https://microsoft.sharepoint.com/teams/msenable/mas/Pages/browse-fixes.aspx

**User Impact:**
Screen reader user unable to know that which dialog is opened.

**MAS Reference:**
MAS1.3.1:
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}
MAS4.2.3:
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={30e4fb40-dcc1-45b4-b714-5e2c28b23ae8}
**Attachment for Reference**

[1959A11y_AzureToolsStorageExplorer_BlobContainerTable_Table_ScreenReader.pptx](https://github.com/microsoft/AzureStorageExplorer/files/3662692/1959A11y_AzureToolsStorageExplorer_BlobContainerTable_Table_ScreenReader.pptx)

**Comment by craxal** [comment]
Same root cause as #1903.

**Comment by MRayermannMSFT** [comment]
@Radha2019 @Charanrajkb @nirmalsahoo @Suraj1200499 @vinodbk197

Please verify whether or not this bug was fixed to your satisfaction in the latest release of Storage Explorer. You can obtain that release from https://www.storageexplorer.com

If the bug has not been fixed to your satisfaction, please:
- Explain how did the fix not meet your expectations.
- Tell us exactly how we can meet your expectations. If there is more than one way to do so, feel free to list the different ways.

**Comment by nirmalsahoo** [comment]
@MRayermannMSFT  Verified the issue on version 1.12.0 and MAC 10.15.3.The issue is still reproduced.

**Comment by craxal** [comment]
@nirmalsahoo VoiceOver appears to be reading what we expect from my end, so I'm not sure what the issue is. Can you share exactly what VoiceOver is reading and what you expect VoiceOver should say?

**Comment by nirmalsahoo** [comment]
@craxal Voiceover should read the row and column reference of the table.Ex: Cell value name with row 1 of Column 1.

**Comment by craxal** [comment]
@nirmalsahoo In other words, you're saying that the screen reader should only read one cell at a time? Does that mean we need to allow the user to select cells with the arrow keys?

We originally added that functionality when we started addressing accessibility issues. But we got called out for not showing visually which cell was selected. However, it doesn't make sense to only highlight the selected cell, because users can't do anything with individual cells. They can only act on entire rows.

**Comment by nirmalsahoo** [comment]
We verified the above issue:
**Actual**issue is: As we are navigating on the Table with Voiceover, Voiceover does not announce column header name with table cells.
**For Ex**: For "blob container table" table cell column header is "Name" but here screen reader not announces column header name along with table cell instead Voiceover announces entire row content in single attempt.

**Expected**: Voiceover should announce column header name, row number and also column number with table cells. And Voiceover should not announce entire row content in single attempt when we navigate to Table cells it should announce table cells individually.
**For Ex**: For "blob container table" table cell column header is "Name", row is 2 and column is 1 here screen reader should announce like: "Accessibility Insights Column header Name Row2 of total no. of rows and column 1 of total no. of columns Column header "Name"  when we navigate to Table cells and it should announce like same information for all table cells individually.
[1959_A11y_AzureToolsStorageExplorer_BlobContainerTable_Table_ScreenReader.zip](https://github.com/microsoft/AzureStorageExplorer/files/4280802/1959_A11y_AzureToolsStorageExplorer_BlobContainerTable_Table_ScreenReader.zip)

**Comment by nirmalsahoo** [comment]
Verified the issue in version 1.12.0 and is still reproduced.
**Actual:**
The voiceover focus does not go to the individual cell of the table and Voiceover does not announce column header name with table cells by using ctrl+options with right/left arrow keys. Voiceover reads as empty application while focus goes to whole table.
**Expected:**
Ex: For "blob container table" table cell column header is "Name", row is 2 and column is 1 here screen reader should announce like: "Accessibility Insights Column header Name Row2 of total no. of rows and column 1 of total no. of columns when we navigate to Table cells and it should announce like same information for all table cells individually.
Please refer the attachment in [one drive](https://microsoft-my.sharepoint.com/:f:/p/v-nisaho/EsSUKgfoOK9HmEL8PqqXd94BUtAVBzKJetpyH6pKAsUxyw?e=bAKDhH) link:

**Comment by craxal** [comment]
@nirmalsahoo We make sure the Name column is announced properly. However, we intentionally made the data table an "application", because we can't use table semantics. This is due to the way the data table is built, which we don't have much control over. We can add a description explaining the use of the arrow keys.

**Comment by MRayermannMSFT** [comment]
Here's what voice over now says:
<img width="736" alt="image" src="https://user-images.githubusercontent.com/21112697/76884369-d6d69380-683a-11ea-9de6-bc0ab1d85c46.png">
Sending build to testers.

**Comment by nirmalsahoo** [comment]
@MRayermannMSFT As per the instruction(use the arrow keys to navigate) to navigate inside the table provided by Voiceover, we are not able to navigate with arrow key inside the table to read out the table data cells corresponding to column header. We have added repro video of issue [here](https://microsoft-my.sharepoint.com/:v:/p/v-nisaho/EVqLSyM93FlMn3vD7lgX6ZsBzYspG33V6SgKg44AVK4EQg?e=N30zU9).

**Comment by nirmalsahoo** [comment]
@MRayermannMSFT @craxal We have verified the bug in below version and still able to reproduce.
<img width="651" alt="Screenshot 2020-03-20 at 3 27 33 PM" src="https://user-images.githubusercontent.com/53940787/77315240-31855900-6d2d-11ea-8f51-0908969a50bb.png">
Could you please help us that we can find the fixes in which version.

**Comment by mstechie** [comment]
GitHubTags:#BM-TCS-StorageExplorer-Mac-Sep2019;Mac;
